packages/contracts/src/contracts/current/protocol/AssetProxy/MixinERC721Transfer.sol

```diff
@@ -19,17 +19,180 @@
 pragma solidity ^0.4.24;
 pragma experimental ABIEncoderV2;
 
+import "./mixins/MAuthorizable.sol";
 import "../../utils/LibBytes/LibBytes.sol";
-import "../../tokens/ERC721Token/ERC721Token.sol";
 import "./libs/LibTransferErrors.sol";
 
 contract MixinERC721Transfer is
-    LibTransferErrors
+    LibTransferErrors,
+    MAuthorizable
 {
     using LibBytes for bytes;
     
     bytes4 constant SAFE_TRANSFER_FROM_SELECTOR = bytes4(keccak256("safeTransferFrom(address,address,uint256,bytes)"));
```

Comment:
We're hard coding the function selectors in the external `transferFrom` functions, but still using variables in `transferFromInternal`. Let's hard code everything to be consistent.